src/components/icons/LottieAnimation.tsx

```diff
@@ -1,38 +1,164 @@
+import { hexToRgb } from '@mui/material'
+import _ from 'lodash'
 import Player, { LottieRefCurrentProps } from 'lottie-react'
-import React, { useEffect, useRef } from 'react'
+import React, { useEffect, useMemo, useRef, useState } from 'react'
+import AnimationKeyframe from '../../@types/lottie/AnimationKeyframe'
+import LottieData from '../../@types/lottie/LottieData'
+import RGBA from '../../@types/lottie/RGBA'
+import ShapeItem from '../../@types/lottie/ShapeItem'
+import ShapeLayer from '../../@types/lottie/ShapeLayer'
 
+/** The props for the LottieAnimation component. */
 interface LottieAnimationProps {
-  animationData: any
+  animationData: LottieData | null
   speed?: number
+  color?: string
   onComplete?: () => void
   style?: React.CSSProperties
 }
 
+/**
+ * Converts hex color to RGBA using Material UI's hexToRgb.
+ *
+ * @param hex - The hex color code.
+ * @returns The RGBA values as [r, g, b, a].
+ */
+const hexToRGBA = (hex: string): [number, number, number, number] => {
+  const rgbString = hexToRgb(hex)
+
+  if (rgbString) {
+    // Regular expression to match rgb or rgba strings
+    const regex = /^rgba?\(\s*(\d{1,3})\s*,\s*(\d{1,3})\s*,\s*(\d{1,3})\s*(?:,\s*([01]?\.?\d*))?\)$/
+    const matches = rgbString.match(regex)
+
+    if (matches) {
+      const r = Number(matches[1])
+      const g = Number(matches[2])
+      const b = Number(matches[3])
+      const a = matches[4] !== undefined ? Number(matches[4]) : 1 // Default alpha to 1 if not present
+      return [r / 255, g / 255, b / 255, a]
+    } else {
+      console.warn('Invalid RGB string format. Falling back to black.')
+    }
+  } else {
+    console.warn('hexToRgb returned null. Falling back to black.')
+  }
+
+  // Fallback to black with full opacity if hex is invalid
+  return [0, 0, 0, 1]
+}
+
+/**
+ * Updates colors in a single shape item.
+ *
+ * This function checks if the item represents a stroke or
+ * fill and updates its color using the provided RGBA array.
+ * If the item contains a nested items array, it recursively
+ * calls itself to ensure that all relevant shapes within the
+ * structure are updated with the new color.
+ *
+ * @param item - The shape item object which may contain
+ * stroke or fill color properties.
+ * @param rgbaArray - The RGBA values to set as new color.
+ */
+const updateColorsInItem = (item: ShapeItem, rgbaArray: RGBA) => {
+  if ((item.ty === 'st' || item.ty === 'fl') && item.c) {
+    // Check for animated color property
+    if (item.c.a === 1) {
+      if (Array.isArray(item.c.k)) {
+        const keyframes = item.c.k as AnimationKeyframe[]
```

Comment:
Why do we need to assert the type here (and why is it safe to assume)? The type of `item.c.k` is `RGBA | number[] | RGBA[] | AnimationKeyframe[]`. How are we sure that the type here will be `AnimationKeyframe[]` and not potentially `RGBA`, `number[]`, or `RGBA[]`? If we know that the type will always be `AnimationKeyframe[]`, perhaps the type for `ShapeItem` needs to be updated accordingly so that we don't need to assert this type here.

Reply:
We need to assert the type because item.c.k can be different types based on whether the color property is static or animated. Earlier, ColorProperty had a broad type for k (RGBA | RGBA[] | number[] | AnimationKeyframe[]) because it was assumed there could be various types during testing and debugging. However, in this context, it's determined that when the color property is animated, item.c.k is specifically an AnimationKeyframe[]. Therefore, it's safe to assert this type here. Adjusting ColorProperty helps handle both static and animated cases appropriately, but the type assertion remains necessary because item.c.k can vary in different instances.

Reply:
I see that the type has been narrowed a bit since that last comment, and the TS compiler assumes that the type of `item.c.k` is `RGBA | AnimationKeyframe[]`. Just trying to avoid 

I'm still not sure if I understand the typing. Is it because `item.ty === 'st' || item.ty === 'fl'` that we can assume the type of `item.c.k` is `AnimationKeyframe[]`? If that's the case, we _could_ update the `ShapeItem` definition to account for that. I'm just hoping to avoid type assertions because making these assumptions (if ever false) can lead to unexpected runtime errors that would have otherwise been caught by TS.

This isn't the only type assertion we have in the project, though, so maybe @raineorshine can give some direction on what is preferable.

Reply:
Thank you for bringing up this concern. You're absolutely right to question whether we can safely assume the type of item.c.k as AnimationKeyframe[] based on item.ty === 'st' || item.ty === 'fl'. Making unchecked assumptions can lead to unexpected runtime errors, especially if the Lottie JSON structure evolves or if we’re working with incomplete or inaccurate typings.
The current assumption is grounded in the Lottie JSON specification. For shape items of type 'st' (stroke) or 'fl' (fill), the c property generally represents color. This property can take two forms:

- A static RGBA array ([r, g, b, a]), or
- A series of animation keyframes (AnimationKeyframe[]) if the animation flag (a) is set to 1.

Thus, the conditional check (item.ty === 'st' || item.ty === 'fl') narrows the scope, making this assumption valid within this specific context. However, as you pointed out, relying on type assertions (item.c.k as AnimationKeyframe[]) can mask potential edge cases or changes in the JSON schema.

Recent Updates:

In the recent commit, I addressed this by refactoring the ColorProperty type into a discriminated union. This ensures that the type of k is correctly tied to the value of a:

```tsx
type ColorProperty =
  | { a: 0; k: RGBA } // Static color
  | { a: 1; k: AnimationKeyframe[] } // Animated color

```

With this structure, TypeScript now guarantees that item.c.k is of type AnimationKeyframe[] whenever item.c.a === 1, eliminating the need for type assertions like item.c.k as AnimationKeyframe[].

The recent changes in the commit enhance type safety by removing unnecessary type assertions and leveraging TypeScript's capabilities. However, input from @raineorshine or other team members on the broader strategy for balancing type safety and development efficiency across the project would be valuable. While type assertions can be acceptable as a temporary solution, they should be mitigated with thorough documentation and robust test coverage. The ultimate goal is to achieve a balance between immediate development needs and long-term maintainability.

Reply:
Thank you so much for your feedback!
I've taken your advice and refactored the code to use type predicates. Specifically:

- Split `ColorProperty` into `StaticColor` and `AnimatedColor` types, each in their own file for better modularity.
- Added the predicate functions `isStaticColor` and `isAnimatedColor` to leverage TypeScript's type narrowing.
- Modified `LottieAnimation.tsx` to utilize the updated types and improved color update logic.